test/unit/gateways/cardprocess_test.rb

```diff
@@ -0,0 +1,280 @@
+require 'test_helper'
+
+class CardprocessTest < Test::Unit::TestCase
+  def setup
+    @gateway = CardprocessGateway.new(user_id: 'login', password: 'password', entity_id: '123')
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      order_id: '1',
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_purchase
+    @gateway.expects(:ssl_post).returns(successful_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal '8a8294495fe8084a016002dd17c163fd', response.authorization
+    assert_equal 'DB', response.params['paymentType']
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    @gateway.expects(:ssl_post).returns(failed_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_failure response
+    assert_equal Gateway::STANDARD_ERROR_CODE[:incorrect_number], response.error_code
+  end
+
+  def test_successful_authorize
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal '8a82944a5fe82704016002caa42c14f8', response.authorization
+    assert_equal 'PA', response.params['paymentType']
+    assert response.test?
+  end
+
+  def test_failed_authorize
+    @gateway.expects(:ssl_post).returns(failed_authorize_response)
+
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_failure response
+    assert_equal 'PA', response.params['paymentType']
+    assert response.test?
+  end
+
+  def test_successful_capture
+    @gateway.expects(:ssl_post).returns(successful_capture_response)
+
+    response = @gateway.capture(@amount, '123', @options)
+    assert_success response
+
+    assert_equal '8a82944a5fe82704016002caa7cd1513', response.authorization
+    assert_equal 'CP', response.params['paymentType']
+    assert response.test?
+  end
+
+  def test_failed_capture
+    @gateway.expects(:ssl_post).returns(failed_capture_response)
+
+    response = @gateway.capture(@money, '123', @options)
```

Comment:
Looks like `@amount` is what is initialized. `@money` is also used on line 88.

Reply:
Thanks for the catch, @deedeelavinder ! This isn't actually why unit tests were failing (that was because XK isn't a country code AM knows about), but it should be fixed anyway. Updated push coming in a second.

Reply:
@bpollack I didn't dig into why the unit tests were throwing errors -- just noticed the discrepancy between `@money` and `@amount`. Glad the tests are easy to fix.